src/main/kotlin/com/feedback/dao/FeedbackDao.kt

```diff
@@ -0,0 +1,75 @@
+package com.feedback.dao
+
+import com.feedback.service.FeedbackFilter
+import org.jooq.*
+import org.springframework.stereotype.Component
+
+@Component
+class FeedbackDao(private val sql: DSLContext) {
+
+    private val feedbackTable = FeedbackTable.instance
+
+    private val retrieveFeedbackTableMapper = RecordMapper<Record, RetrievedFeedback> { record ->
+        RetrievedFeedback(
+            id = record[feedbackTable.id],
+            department = record[feedbackTable.department]?.let { DepartmentType.fromString(it) },
+            comment = record[feedbackTable.comment],
+            creationTimestamp = record[feedbackTable.creationTimestamp],
+            status = record[feedbackTable.status]?.let  { StatusType.fromString(it) } ?: StatusType.NOT_SOLVED, //todo ask why can be nullable
+            employeeId = record[feedbackTable.employeeId]
+        )
+    }
+
+    fun insertFeedback(userDetails: LoggedInUser, feedback: InsertFeedback): Long {
```

Comment:
InsertFeedback isn't the best name for that data class. I will call it: FeedbackCreationRequest.

Reply:
fixed, good point